vms/platformvm/txs/executor/state_changes_test.go

```diff
@@ -214,3 +226,142 @@ func TestAdvanceTimeTo_RemovesStaleExpiries(t *testing.T) {
+				ValidationID:      ids.GenerateTestID(),
+				SubnetID:          ids.GenerateTestID(),
+				NodeID:            ids.GenerateTestNodeID(),
+				PublicKey:         pkBytes,
+				Weight:            1,
+				EndAccumulatedFee: endAccumulatedFee,
+			}
+		}
+		sovToEvict0 = newSoV(3 * units.NanoAvax) // lasts 3 seconds
+		sovToEvict1 = newSoV(3 * units.NanoAvax) // lasts 3 seconds
+		sovToKeep   = newSoV(units.Avax)
+
+		currentTime = genesistest.DefaultValidatorStartTime
+		newTime     = currentTime.Add(timeToAdvance)
+
+		config = config.Config{
+			ValidatorFeeConfig: fee.Config{
+				Capacity:                 genesis.LocalParams.ValidatorFeeConfig.Capacity,
+				Target:                   1,
+				MinPrice:                 genesis.LocalParams.ValidatorFeeConfig.MinPrice,
+				ExcessConversionConstant: genesis.LocalParams.ValidatorFeeConfig.ExcessConversionConstant,
+			},
+			UpgradeConfig: upgradetest.GetConfig(upgradetest.Latest),
+		}
+	)
+
+	tests := []struct {
+		name             string
+		initialSoVs      []state.SubnetOnlyValidator
+		expectedModified bool
+		expectedSoVs     []state.SubnetOnlyValidator
+		expectedExcess   gas.Gas
+	}{
+		{
+			name:             "no SoVs",
+			expectedModified: false,
+			expectedExcess:   0,
+		},
+		{
+			name: "evicted one",
+			initialSoVs: []state.SubnetOnlyValidator{
+				sovToEvict0,
+			},
+			expectedModified: true,
+			expectedExcess:   0,
+		},
+		{
+			name: "evicted multiple",
+			initialSoVs: []state.SubnetOnlyValidator{
+				sovToEvict0,
+				sovToEvict1,
+			},
+			expectedModified: true,
+			expectedExcess:   3,
+		},
+		{
+			name: "evicted only 2",
+			initialSoVs: []state.SubnetOnlyValidator{
+				sovToEvict0,
+				sovToEvict1,
+				sovToKeep,
+			},
+			expectedModified: true,
+			expectedSoVs: []state.SubnetOnlyValidator{
+				sovToKeep,
+			},
+			expectedExcess: 6,
+		},
+		{
+			name: "chooses not to evict",
+			initialSoVs: []state.SubnetOnlyValidator{
+				sovToKeep,
+			},
+			expectedModified: false,
+			expectedSoVs: []state.SubnetOnlyValidator{
+				sovToKeep,
+			},
+			expectedExcess: 0,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				require = require.New(t)
+				s       = statetest.New(t, statetest.Config{})
+			)
+
+			for _, sov := range test.initialSoVs {
+				require.NoError(s.PutSubnetOnlyValidator(sov))
+			}
+
+			// Ensure the invariant that [newTime <= nextStakerChangeTime] on
+			// AdvanceTimeTo is maintained.
+			nextStakerChangeTime, err := state.GetNextStakerChangeTime(
+				genesis.LocalParams.ValidatorFeeConfig,
+				s,
+				mockable.MaxTime,
+			)
+			require.NoError(err)
+			require.False(newTime.After(nextStakerChangeTime))
+
+			validatorsModified, err := AdvanceTimeTo(
+				&Backend{
+					Config: &config,
+				},
+				s,
+				newTime,
+			)
+			require.NoError(err)
+			require.Equal(test.expectedModified, validatorsModified)
+
+			activeSoVs, err := s.GetActiveSubnetOnlyValidatorsIterator()
```

Comment:
should this iterator be released?

Reply:
`iterator.ToSlice` consumes the iterator (releases it).